apps/challenges/aws_utils.py

```diff
@@ -1109,6 +1110,251 @@ def create_eks_nodegroup(challenge, cluster_name):
     create_service_by_challenge_pk(client, challenge_obj, client_token)
 
 
+@app.task
+def setup_eks_cluster(challenge):
+    """
+    Creates EKS and NodeGroup ARN roles
+
+    Arguments:
+        instance {<class 'django.db.models.query.QuerySet'>} -- instance of the model calling the post hook
+    """
+    from .models import ChallengeEvaluationCluster
+    from .serializers import ChallengeEvaluationClusterSerializer
+    from .utils import get_aws_credentials_for_challenge
+
+    for obj in serializers.deserialize("json", challenge):
+        challenge_obj = obj.object
+    challenge_aws_keys = get_aws_credentials_for_challenge(challenge_obj.pk)
+    client = get_boto3_client("iam", challenge_aws_keys)
+    eks_role_name = "evalai-code-upload-eks-role"
```

Comment:
Can we please attach challenge pk along with `staging`, `production` keyword to it so that it is unique for every challenge and we know exactly which challenge it is evaluating.